micro-rdk/src/esp32/entry.rs

```diff
@@ -205,7 +206,11 @@ pub async fn serve_web_inner(
         .with_webrtc(webrtc)
         .with_periodic_app_client_task(Box::new(RestartMonitor::new(|| unsafe {
             crate::esp32::esp_idf_svc::sys::esp_restart()
-        })));
+        })))
+        .with_periodic_app_client_task(Box::new(ConfigMonitor::new(||unsafe {
```

Comment:
I expect this review to land after #251 which will add config caching. If we detect a config change, consider whether we should write the new config to NVS before restarting. That way, if we happen to come back up and find we can't talk to WiFi anymore, the robot will still start up with the newest configuration.

Reply:
Sounds like a plan, should this be in a new ticket? Not 100% sure about ticket splitting rules but would feel nice to have this PR closed with a working config restart before looking into NVS update

Reply:
Yes, I'd suggest filing a new ticket, and then making it depend on the ticket for this PR and the ticket for #251 .

Reply:
FYI #251 merged, so you will likely have some merge conflicts to deal with before you can merge this.

Reply:
Also, please post the new ticket here after you file it.

Reply:
https://viam.atlassian.net/browse/RSDK-8159

Done!